src/test/java/org/javarosa/core/model/DynamicSelectUpdateTest.java

```diff
@@ -0,0 +1,323 @@
+
+        scenario.answer("/data/repeat[0]/value","a");
+        scenario.answer("/data/repeat[0]/label","A");
+        assertThat(scenario.choicesOf("/data/select"), contains(
+            choice("a", "A")));
+
+        scenario.answer("/data/repeat[0]/value","c");
+        scenario.answer("/data/repeat[0]/label","C");
+        assertThat(scenario.choicesOf("/data/select"), contains(
+            choice("c", "C")));
+        assertThat(scenario.choicesOf("/data/select").size(), is(1));
+    }
+
+    @Test
+    public void selectFromRepeat_whenRepeatRemoved_updatesChoices() throws Exception {
+        Scenario scenario = Scenario.init("Select from repeat", getSelectFromRepeatForm());
+
+        scenario.answer("/data/repeat[0]/value", "a");
+        scenario.answer("/data/repeat[0]/label", "A");
+        assertThat(scenario.choicesOf("/data/select"), contains(
+            choice("a", "A")));
+
+        scenario.removeRepeat("/data/repeat[0]");
+        assertThat(scenario.choicesOf("/data/select").size(), is(0));
+    }
+
+    @Test
+    public void selectFromRepeat_withPredicate_whenPredicateTriggerChanges_updatesChoices() throws Exception {
+        Scenario scenario = Scenario.init("Select from repeat", getSelectFromRepeatForm("starts-with(value,current()/../filter)"));
+
+        scenario.answer("/data/repeat[0]/value", "a");
+        scenario.answer("/data/repeat[0]/label", "A");
+        scenario.answer("/data/filter", "a");
+        assertThat(scenario.choicesOf("/data/select"), contains(
+            choice("a", "A")));
+
+        scenario.answer("/data/filter", "b");
+        assertThat(scenario.choicesOf("/data/select").size(), is(0));
+    }
+
+    private static XFormsElement getSelectFromRepeatForm() {
+        return getSelectFromRepeatForm("");
+    }
+
+    private static XFormsElement getSelectFromRepeatForm(String predicate) {
+        return html(
+            head(
+                title("Select from repeat"),
+                model(
+                    mainInstance(
+                        t("data id=\"repeat-select\"",
+                            t("repeat",
+                                t("value"),
+                                t("label")),
+                            t("select"),
+                            t("filter"))))),
+            body(
+                repeat("/data/repeat",
+                    input("value"),
+                    input("label")),
+                select1Dynamic("/data/select", "../repeat" + (!predicate.isEmpty() ? "[" + predicate + "]" : "")),
+                input("filter")
+            )
+        );
+    }
+    //endregion
+
+    //region Select in repeat
+    // When a dynamic select is in a repeat, the itemsets for all repeat instances are represented by the same ItemsetBinding.
+    @Test
+    public void selectInRepeat_withPredicate_evaluatesChoiceListForEachRepeatInstance() throws Exception {
+        Scenario scenario = Scenario.init("Select from repeat", html(
```

Comment:
Did you mean for the form name to be `Select in repeat`?

Reply:
Ah yes, indeed!